### PR TITLE
Add two-factor authentication (TOTP)

### PR DESCRIPTION
**Pull request by mworrell**
### Description

This adds a 2FA passcode to the logon flow.

The passcode-secret can be scanned using Google Authenticator or another TOTP App.

### TODO

 - [x] Add barcode to scan
 - [x] Add passcode to log on form fields
 - [x] Check passcode, error feedback
 - [x] Admin panel for setting/resetting 2FA per user
 - [x] Admin page to set 2FA mode for site and per user group
 - ~Reset 2FA code after password reset~
 - [x] Check 2FA on password reset form (not on send reminder, as we don't want to expose the available users)
 - [x] Check texts with stake holders
 - [x] Translate strings

### Checklist

- [ ] documentation updated
- [ ] tests added
- [x] no BC breaks

### CODE REVIEW

**Comment by mworrell** [comment]
@ddeboer Could you check this?  (Enable mod_auth2fa and check admin-person pages and login/password-reset flows.)

**Comment by ddeboer** [comment]
Improved labels and translations a bit, with links to authentication apps.

File: modules/mod_auth2fa/translations/nl.po [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+"Content-Type: text/plain; charset=utf-8\n"
+"Content-Transfer-Encoding: 8bit\n"
+"Last-Translator: Marc Worrell <marc@worrell.nl>\n"
+"Language-Team: \n"
+"Language: nl\n"
+"X-Generator: Poedit 2.2.1\n"
+
+#: modules/mod_auth2fa/mod_auth2fa.erl:68
+msgid "Add two-factor authentication"
+msgstr "Voeg twee-factor-authenticatie toe"
```
[the remark this answers is not in the document]
Comment (ddeboer) [reply modules/mod_auth2fa/translations/nl.po]: I decided to adopt this translation in most places.

[the remark this answers is not in the document]
Comment (mworrell) [reply modules/mod_auth2fa/translations/nl.po]: I fetched that (dashed) translation from the Apple documentation. I guessed they have thought hard about their texts.

File: modules/mod_auth2fa/mod_auth2fa.erl [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+
+-module(mod_auth2fa).
+-author("Marc Worrell <marc@worrell.nl>").
+
+-mod_title("Authentication 2FA").
+-mod_description("Add 2FA authentication using TOTP").
```
[the remark this answers is not in the document]
Comment (mworrell) [reply modules/mod_auth2fa/mod_auth2fa.erl]: Agree.

File: modules/mod_auth2fa/translations/nl.po (hunk 1 above)
[the remark this answers is not in the document]
Comment (ddeboer) [reply modules/mod_auth2fa/translations/nl.po]: Ah, that’s interesting! They even [differentiate](https://support.apple.com/nl-nl/HT207198) between [the two](https://support.apple.com/nl-nl/HT204152). Let me think about this some more.

[the remark this answers is not in the document]
Comment (mworrell) [reply modules/mod_auth2fa/translations/nl.po]: Interesting..  "twee-staps-verificatie" and "twee-factor-authenticatie". We can merge for now, and then later adapt the texts after checking with @michielklonhammer as well.

[the remark this answers is not in the document]
Comment (mworrell) [reply modules/mod_auth2fa/translations/nl.po]: Ah "twee-factor" is used in for TOTP and the like.  And "twee-staps" is used for sending a text messages with a code. So I think we can stick with the "twee-factor" translation.

**Pull request merged**